### PR TITLE
SNOW-683056 Increase test coverage for type_utils.py

### PR DESCRIPTION
**Pull request by sfc-gh-sfan**
Description

Added/Modified tests to increase test coverage. Also removed some unused code path.

Testing

unit/integ tests

Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-683056

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Added/Modified tests to increase test coverage. Also removed some unused code path.

### CODE REVIEW

File: src/snowflake/snowpark/_internal/type_utils.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def convert_sf_to_sp_type(
         return VariantType()
     if column_type_name == "OBJECT":
         return MapType(StringType(), StringType())
-    if column_type_name == "GEOGRAPHY":  # not supported by python connector
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/_internal/type_utils.py:80]: I believe this is supported now, but happy to revert if it is not the case.

File: src/snowflake/snowpark/_internal/type_utils.py [hunk 2]
```diff
@@ -528,18 +528,16 @@ def visit_ClassDef(self, node):
 
 # Get a mapping from type string to type object, for cast() function
 def get_data_type_string_object_mappings(
-    to_fill_dict: Optional[Dict[str, Type[DataType]]] = None,
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/_internal/type_utils.py:531]: This function only has one call site which provides `to_fill_dict`, so it does not seem reasonable to be more flexible than that.

**Review by sfc-gh-stan — APPROVED** [review]

File: tests/integ/test_dataframe.py [hunk 3]
```diff
@@ -1135,29 +1135,48 @@ def test_create_dataframe_with_basic_data_types(session):
...
-        array("I", [1, 2, 3]),
-        {"'": 1},
+        [
+            ["'", 2],
+            ("'", 2),
+            [[1, 2], [2, 1]],
+            array("I", [1, 2, 3]),
+            {"'": 1},
+        ],
+        [
```
Comment (sfc-gh-jdu) [inline tests/integ/test_dataframe.py:1145]: why do we need another row here?

**Review by sfc-gh-jdu — APPROVED** [review]

File: tests/integ/test_dataframe.py (hunk 3 above)
Comment (sfc-gh-sfan) [reply tests/integ/test_dataframe.py:1145]: This is mainly to make sure merge type logic is covered for complex type: https://github.com/snowflakedb/snowpark-python/blob/8c52606a44096a265ddbdcd4d514e1ca8d51b9a1/src/snowflake/snowpark/_internal/type_utils.py#L320

The coverage report said that the `MapType` is not covered, and it feels like this is the easiest fix :)

**Pull request merged**